test/spec/modules/greenbidsBidAdapter_specs.js

```diff
@@ -0,0 +1,964 @@
+      const payload = JSON.parse(request.data);
+
+      expect(payload.pageDescription).to.exist;
+      expect(payload.pageDescription).to.deep.equal(testText);
+    });
+
+    it('should add pageDescription info to payload based on open-graph description', function () {
+      const testText = 'This is a description from open-graph';
+      sandbox.stub(window.top.document, 'querySelector').withArgs('meta[property="og:description"]').returns({ content: testText });
+
+      const request = spec.buildRequests(bidRequests, bidderRequestDefault);
+      const payload = JSON.parse(request.data);
+
+      expect(payload.pageDescription).to.exist;
+      expect(payload.pageDescription).to.deep.equal(testText);
+    });
+
+    it('should add pageDescription info to payload sliced on 300 first characters', function () {
+      const testText = Array(500).join('a');
+      sandbox.stub(window.top.document, 'querySelector').withArgs('meta[name="description"]').returns({ content: testText });
+
+      const request = spec.buildRequests(bidRequests, bidderRequestDefault);
+      const payload = JSON.parse(request.data);
+
+      expect(payload.pageDescription).to.exist;
+      expect(payload.pageDescription).to.have.length(300);
+    });
+
+    it('should add pageDescription info to payload when fallbacking from window.top', function () {
+      const testText = 'This is a fallback description';
+      sandbox.stub(window.top.document, 'querySelector').throws();
+      sandbox.stub(document, 'querySelector').withArgs('meta[name="description"]').returns({ content: testText });
+
+      const request = spec.buildRequests(bidRequests, bidderRequestDefault);
+      const payload = JSON.parse(request.data);
+
+      expect(payload.pageDescription).to.exist;
+      expect(payload.pageDescription).to.deep.equal(testText);
+    });
+
+    it('should add timeToFirstByte info to payload', function () {
+      const request = spec.buildRequests(bidRequests, bidderRequestDefault);
+      const payload = JSON.parse(request.data);
+      const performance = window.performance || window.webkitPerformance || window.msPerformance || window.mozPerformance;
+
+      const ttfbExpectedV2 = performance &&
+        typeof performance.getEntriesByType === 'function' &&
+        Object.prototype.toString.call(performance.getEntriesByType) === '[object Function]' &&
+        performance.getEntriesByType('navigation')[0] &&
+        performance.getEntriesByType('navigation')[0].responseStart &&
+        performance.getEntriesByType('navigation')[0].requestStart &&
+        performance.getEntriesByType('navigation')[0].responseStart > 0 &&
+        performance.getEntriesByType('navigation')[0].requestStart > 0 &&
+        Math.round(
+          performance.getEntriesByType('navigation')[0].responseStart - performance.getEntriesByType('navigation')[0].requestStart
+        );
+
+      expect(payload.timeToFirstByte).to.exist;
```

Comment:
suggestion: here you clone the logic in the actual code, I'm not sure it's the right way to test it... Could you mock the `performance` to avoid all these checks here ?